py-polars/polars/dataframe/frame.py

```diff
@@ -3234,18 +3230,19 @@ def write_excel(
 
         for column in df.columns:
             col_idx, options = table_start[1] + df.get_column_index(column), {}
+            fmt = column_formats.get(column)
             if column in hidden_columns:
                 options = {"hidden": True}
             if column in column_widths:  # type: ignore[operator]
                 ws.set_column_pixels(
                     col_idx,
                     col_idx,
                     column_widths[column],  # type: ignore[index]
-                    None,
+                    fmt,
                     options,
                 )
             elif options:
-                ws.set_column(col_idx, col_idx, None, None, options)
```

Comment:
This should be a little cleaner (and still apply the format if there is no option/width):
```python
for c in df.columns:
    col_idx, options = table_start[1] + df.get_column_index(c), {}
    if c in hidden_columns:
        options = {"hidden": True}

    if options or c in column_formats c in column_widths:
        ws.set_column(
            col_idx,
            col_idx,
            column_widths.get(c),
            column_formats.get(c),
            options,
        )`